opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -1888,6 +2064,9 @@ namespace Opm {
         // network related
         bool more_network_update = false;
         if (shouldBalanceNetwork(episodeIdx, iterationIdx) || mandatory_network_balance) {
+            const double dt = this->ebosSimulator_.timeStepSize();
+            // Calculate common THP for subsea manifold well group (item 3 of NODEPROP set to YES)
+            computeWellGroupThp(dt, deferred_logger);
```

Comment:
I believe `computeWellGroupThp()` should be called within the function. Is there some practical reasons we have to call them in this file? for example, the need to access the simulator?

Reply:
I recall I had a look  at this before and run into issues as the one you mentioned. I can have second try and investigate it. Maybe we could have a chat about this?

Reply:
Sure. I think some refactoring can be done here. Let me review it first.